internal/evidence/reactor.go

```diff
@@ -101,6 +101,18 @@ func (r *Reactor) handleEvidenceMessage(ctx context.Context, envelope *p2p.Envel
 
 	switch msg := envelope.Message.(type) {
 	case *tmproto.Evidence:
+
+		// Only accept evidence if we are an active validator.
+		// On other hosts, signatures in evidence (if any) cannot be verified due to lack of validator public keys,
+		// and it creates risk of adding invalid evidence to the pool.
+		//
+		// TODO: We need to figure out how to handle evidence from non-validator nodes, to avoid scenarios where some
+		// evidence is lost.
+		if !r.evpool.state.Validators.HasPublicKeys {
+			// silently drop the message
+			logger.Debug("dropping evidence message as we are not a validator", "evidence", envelope.Message)
```

Comment:
I guess we need to return nil here or somehow to prevent further execution of the function.

Reply:
fixed